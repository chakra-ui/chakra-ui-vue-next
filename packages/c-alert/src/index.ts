import { defineComponent, computed, createVNode, mergeProps, isVNode, Fragment } from 'vue';
import { createStylesContext, useMultiStyleConfig, chakra } from '@chakra-ui/vue-system';
import { createContext, getValidChildren } from '@chakra-ui/vue-utils';
import { createIconComponent } from '@chakra-ui/c-icon';

function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };
  return _extends.apply(this, arguments);
}

var CInfoIcon = createIconComponent("info");
var CCheckIcon = createIconComponent("check-circle");
var CWarningIcon = createIconComponent("warning-alt");
var CErrorIcon = createIconComponent("warning");

function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !isVNode(s);
}

var STATUSES = {
  info: {
    colorScheme: "blue",
    icon: CInfoIcon
  },
  success: {
    colorScheme: "green",
    icon: CCheckIcon
  },
  warning: {
    colorScheme: "orange",
    icon: CWarningIcon
  },
  error: {
    colorScheme: "red",
    icon: CErrorIcon
  },
  loading: {
    icon: CInfoIcon,
    colorScheme: "blue"
  }
};

var _createStylesContext = createStylesContext("Alert"),
    StylesProvider = _createStylesContext[0],
    useStyles = _createStylesContext[1];

var _createContext = createContext({
  name: "AlertContext",
  errorMessage: "useAlertContext: `context` is undefined. Seems you forgot to wrap alert components in `<c-alert />`"
}),
    AlertProvider = _createContext[0],
    useAlertContext = _createContext[1];
/**
 * CAlert component
 *
 * This is the container component for all Alert components.
 * It also provides state and context to it's compound components
 */


var CAlert = defineComponent({
  name: "CAlert",
  props: {
    as: {
      type: [String, Object],
      "default": "div"
    },
    status: {
      type: [String],
      "default": "info"
    },
    colorScheme: {
      type: [String]
    },
    styleConfig: {
      type: [Object]
    },
    variant: {
      type: [String],
      "default": "solid"
    }
  },
  setup: function setup(props, _ref) {
    var slots = _ref.slots,
        attrs = _ref.attrs;
    var colorScheme = computed(function () {
      var _STATUSES$props$statu;

      return props.colorScheme || ((_STATUSES$props$statu = STATUSES[props == null ? void 0 : props.status]) == null ? void 0 : _STATUSES$props$statu.colorScheme);
    });
    var themingProps = computed(function () {
      return {
        colorScheme: colorScheme.value,
        variant: props.variant
      };
    });
    AlertProvider({
      status: computed(function () {
        return props.status;
      })
    });
    var styles = useMultiStyleConfig("Alert", themingProps);
    StylesProvider(styles);
    var alertStyles = computed(function () {
      return _extends({
        width: "100%",
        display: "flex",
        alignItems: "center",
        position: "relative",
        overflow: "hidden"
      }, styles.value.container);
    });
    return function () {
      return createVNode(chakra.div, mergeProps({
        "role": "alert",
        "__label": "alert",
        "__css": alertStyles.value
      }, attrs), {
        "default": function _default() {
          return getValidChildren(slots);
        }
      });
    };
  }
});
/**
 * CAlertTitle component
 *
 * The title component for alerts
 */

var CAlertTitle = defineComponent({
  name: "CAlertTitle",
  setup: function setup(_, _ref2) {
    var attrs = _ref2.attrs,
        slots = _ref2.slots;
    var styles = useStyles();
    return function () {
      return createVNode(chakra.div, mergeProps({
        "__label": "alert__title",
        "__css": styles.value.title
      }, attrs), _isSlot(slots) ? slots : {
        "default": function _default() {
          return [slots];
        }
      });
    };
  }
});
/**Â«
 * CAlertDescription component
 *
 * The description component for alerts
 */

var CAlertDescription = defineComponent({
  name: "CAlertDescription",
  setup: function setup(_, _ref3) {
    var attrs = _ref3.attrs,
        slots = _ref3.slots;
    var styles = useStyles();
    return function () {
      return createVNode(chakra.div, mergeProps({
        "__label": "alert__description",
        "__css": styles.value.description
      }, attrs), {
        "default": function _default() {
          return getValidChildren(slots);
        }
      });
    };
  }
});
/**
 * CAlertIcon component
 *
 * The Icon component for alerts
 */

var CAlertIcon = defineComponent({
  name: "CAlertIcon",
  setup: function setup(_, _ref4) {
    var attrs = _ref4.attrs,
        slots = _ref4.slots;
    var styles = useStyles();

    var _useAlertContext = useAlertContext(),
        status = _useAlertContext.status;

    var BaseIcon = STATUSES[status.value].icon;
    var css = computed(function () {
      return status.value === "loading" ? styles.value.spinner : styles.value.icon;
    });
    return function () {
      var validChildren = getValidChildren(slots);
      return createVNode(chakra.span, mergeProps({
        "display": "inherit",
        "__label": "alert__icon"
      }, attrs, {
        "__css": css.value
      }), {
        "default": function _default() {
          return createVNode(Fragment, null, [validChildren.length ? slots : createVNode(BaseIcon, {
            "h": "100%",
            "w": "100%"
          }, null)]);
        }
      }); // return <icon {...styles.value.icon} {...attrs}></icon>
    };
  }
});

export { CAlert, CAlertDescription, CAlertIcon, CAlertTitle };
