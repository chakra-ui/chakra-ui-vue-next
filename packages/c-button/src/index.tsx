import { defineComponent } from "vue"

export { default as CButton } from "./button"
export { default as CButtonGroup } from "./button-group"
export { default as CIconButton } from "./icon-button"
