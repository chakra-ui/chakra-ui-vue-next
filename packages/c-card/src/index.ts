export { CCard, type CCardProps } from "./c-card"
export { CCardBody, type CCardBodyProps } from "./c-card-body"
export { CCardHeader, type CCardHeaderProps } from "./c-card-header"
