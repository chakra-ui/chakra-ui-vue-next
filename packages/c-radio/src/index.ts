export { CRadioGroup, type CRadioGroupProps } from "./c-radio-group"
export { CRadio, type CRadioProps } from "./c-radio"
export type { RadioGroupContext } from "./radio-context"
