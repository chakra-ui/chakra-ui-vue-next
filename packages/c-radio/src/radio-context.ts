import * as CSS from "csstype"
import * as radio from "@zag-js/radio-group"
import { createContext } from "@chakra-ui/vue-utils"
import type { UseRadioGroupReturn } from "./use-radio-group"
import { UnwrapRef } from "vue"
import { createStylesContext } from "@chakra-ui/vue-system"
import * as VS from "@chakra-ui/vue-system"

export type RadioContext = Parameters<
  ReturnType<typeof radio.connect>["getRadioProps"]
>[0]

export const [RadioProvider, useRadioContext] = createContext<RadioContext>({
  name: "CRadioContext",
  strict: true,
})

export type RadioGroupContext = UnwrapRef<UseRadioGroupReturn>

export const [RadioGroupProvider, useRadioGroupContext] =
  createContext<UseRadioGroupReturn>({
    name: "CRadioGroupContext",
    strict: true,
  })

export const [RadioGroupStylesProvider, useRadioGroupStyles] =
  createStylesContext("CRadioGroup")
