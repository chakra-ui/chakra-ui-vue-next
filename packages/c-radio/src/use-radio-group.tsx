import { computed, getCurrentInstance, reactive } from "vue"
import { normalizeProps, useMachine } from "@zag-js/vue"
import * as radio from "@zag-js/radio-group"
import { useId } from "@chakra-ui/vue-composables"
import { Optional } from "@chakra-ui/vue-utils"

export type UseRadioGroupProps = Optional<radio.Context, "id"> & {
  modelValue?: radio.Context["value"]
}

export const useRadioGroup = (props: UseRadioGroupProps) => {
  const reactiveProps = reactive(props)
  const instance = getCurrentInstance()

  const [state, send] = useMachine(
    radio.machine({
      ...reactiveProps,
      id: useId().value,
      value: reactiveProps.modelValue ?? reactiveProps.value,
      onChange(details) {
        instance?.emit("change", details.value)
        instance?.emit("update:modelValue", details.value)
      },
    })
  )

  const api = computed(() => radio.connect(state.value, send, normalizeProps))

  return api
}

export type UseRadioGroupReturn = ReturnType<typeof useRadioGroup>
