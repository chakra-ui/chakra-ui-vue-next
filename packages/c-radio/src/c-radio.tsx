/**
 * Hey! Welcome to @chakra-ui/vue-next CRadio
 *
 * Radios are used when only one choice may be selected in a series of options
 *
 * @see Docs     https://next.vue.chakra-ui.com/c-radio
 * @see Source   https://github.com/chakra-ui/chakra-ui-vue-next/blob/main/packages/c-radio/src/c-radio/c-radio.ts
 * @see WAI-ARIA https://www.w3.org/TR/wai-aria-practices-1.2
 */

import {
  computed,
  defineComponent,
  h,
  InputHTMLAttributes,
  mergeProps,
  PropType,
  reactive,
} from "vue"
import {
  chakra,
  HTMLChakraProps,
  type ThemingProps,
  type StyleResolverProps,
  useMultiStyleConfig,
} from "@chakra-ui/vue-system"
import * as VS from "@chakra-ui/vue-system"
import { getValidChildren, SNAO, vueThemingProps } from "@chakra-ui/vue-utils"
import { RadioContext, useRadioGroupContext } from "./radio-context"

export interface CRadioProps
  extends Omit<HTMLChakraProps<"label">, keyof RadioContext>,
    ThemingProps<"Radio">,
    RadioContext {
  /**
   * The spacing between the checkbox and its label text
   * @default 0.5rem
   * @type SystemProps["marginLeft"]
   */
  spacing?: StyleResolverProps["marginLeft"]
  /**
   * Additional props to be forwarded to the `input` element
   */
  inputProps?: InputHTMLAttributes
}

export const CRadio = defineComponent({
  props: {
    value: {
      type: String as PropType<CRadioProps["value"]>,
      required: true,
    },
    disabled: {
      type: Boolean as PropType<CRadioProps["disabled"]>,
    },
    invalid: {
      type: Boolean as PropType<CRadioProps["invalid"]>,
    },
    readOnly: {
      type: Boolean as PropType<CRadioProps["readOnly"]>,
    },
    spacing: {
      type: SNAO as PropType<CRadioProps["spacing"]>,
      default: "0.5rem",
    },
    ...vueThemingProps,
  },
  setup(props, { slots, attrs }) {
    const groupApi = useRadioGroupContext()

    const styleAttrs = computed(() => mergeProps(props, groupApi.value, attrs))

    const styles = useMultiStyleConfig("Radio", styleAttrs)

    const radioProps = reactive({
      value: props.value,
      disabled: props.disabled,
      invalid: props.invalid,
      readOnly: props.readOnly,
    })

    const inputProps = computed(() => {
      const apiInputProps = groupApi.value.getRadioInputProps(radioProps)
      const apiInputState = groupApi.value.getRadioState(radioProps)

      return {
        ...apiInputProps,
        // modelValue: apiInputState.isChecked,
      }
    })

    const rootStyles = computed(() => ({
      display: "inline-flex",
      alignItems: "center",
      verticalAlign: "top",
      cursor: "pointer",
      position: "relative",
      ...styles.value.container,
    }))

    const labelStyles = computed(() => ({
      userSelect: "none",
      marginStart: props.spacing,
      ...styles.value.label,
    }))

    const controlStyles = computed(() => ({
      display: "inline-flex",
      alignItems: "center",
      justifyContent: "center",
      flexShrink: 0,
      ...styles.value.control,
    }))

    return () => (
      <chakra.label
        {...groupApi.value.getRadioProps(radioProps)}
        __css={rootStyles.value}
        {...attrs}
      >
        <chakra.input
          __chakraIsRaw
          __label="radio__input"
          {...inputProps.value}
        />
        <chakra.span
          __label="radio__control"
          {...groupApi.value.getRadioControlProps(radioProps)}
          __css={controlStyles.value}
        />
        <chakra.span
          __label="radio__label"
          {...groupApi.value.getRadioLabelProps(radioProps)}
          __css={labelStyles.value}
        >
          {() => getValidChildren(slots)}
        </chakra.span>
      </chakra.label>
    )
  },
})
