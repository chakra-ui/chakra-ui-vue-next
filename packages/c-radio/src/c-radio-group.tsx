import { h, PropType, defineComponent } from "vue"
import {
  HTMLChakraProps,
  ThemingProps,
  chakra,
  useMultiStyleConfig,
} from "@chakra-ui/vue-system"
import { useThemingProps, vueThemingProps } from "@chakra-ui/vue-utils"
import { UseRadioGroupProps, useRadioGroup } from "./use-radio-group"
import { RadioGroupProvider, RadioGroupStylesProvider } from "./radio-context"

export interface CRadioGroupProps
  extends UseRadioGroupProps,
    HTMLChakraProps<"div">,
    Omit<ThemingProps<"Radio">, "orientation"> {}

export const CRadioGroup = defineComponent({
  name: "CRadioGroup",
  props: {
    dir: {
      type: String as PropType<CRadioGroupProps["dir"]>,
    },
    disabled: {
      type: Boolean as PropType<CRadioGroupProps["disabled"]>,
    },
    form: {
      type: String as PropType<CRadioGroupProps["form"]>,
    },
    getRootNode: {
      type: Function as PropType<CRadioGroupProps["getRootNode"]>,
    },
    id: {
      type: String as PropType<CRadioGroupProps["id"]>,
    },
    ids: {
      type: Object as PropType<CRadioGroupProps["ids"]>,
    },
    name: {
      type: String as PropType<CRadioGroupProps["name"]>,
    },
    modelValue: {
      type: String as PropType<CRadioGroupProps["modelValue"]>,
    },
    orientation: {
      type: String as PropType<CRadioGroupProps["orientation"]>,
    },
    readonly: {
      type: Boolean as PropType<CRadioGroupProps["readOnly"]>,
    },
    ...vueThemingProps,
  },
  emits: ["change", "update:modelValue"],
  setup(props, { slots, attrs, expose }) {
    const api = useRadioGroup(props)

    const themeProps = useThemingProps(props)

    const styles = useMultiStyleConfig("Radio", themeProps)

    RadioGroupProvider(api)

    RadioGroupStylesProvider(styles)

    expose(api.value)

    return () => (
      <chakra.div __label="radio-group" {...api.value.rootProps} {...attrs}>
        {slots.default?.(api.value)}
      </chakra.div>
    )
  },
})
