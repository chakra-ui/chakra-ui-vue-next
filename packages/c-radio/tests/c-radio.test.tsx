import { defineComponent, h } from "vue"
import { render } from "@chakra-ui/vue-test-utils/src"
import { CRadio, CRadioGroup } from "../src"

it("should render properly", () => {
  const { asFragment } = render({
    components: { CRadioGroup, CRadio },
    template: `
      <CRadioGroup>
        <CRadio value="1">One</CRadio>
        <CRadio value="2">Two</CRadio>
        <CRadio value="3">Three</CRadio>
      </CRadioGroup>
    `,
  })
  expect(asFragment()).toMatchSnapshot()
})
