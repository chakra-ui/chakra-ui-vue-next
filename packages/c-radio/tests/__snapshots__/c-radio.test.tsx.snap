// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="radio-group:1:label"
    class="chakra-radio-group chakra-0"
    data-part="root"
    data-scope="radio-group"
    id="radio-group:1"
    role="radiogroup"
  >
    <label
      class="chakra-1cqh9jq"
      data-part="radio"
      data-scope="radio-group"
      for="radio-group:1:radio:input:1"
      id="radio-group:1:radio:1"
    >
      <input
        class="chakra-radio__input chakra-0"
        data-ownedby="radio-group:1"
        data-part="radio-input"
        data-scope="radio-group"
        id="radio-group:1:radio:input:1"
        name="1"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap; word-wrap: normal;"
        type="radio"
        value="1"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control chakra-vwtg5w"
        data-part="radio-control"
        data-scope="radio-group"
        id="radio-group:1:radio:control:1"
      />
      <span
        class="chakra-radio__label chakra-4mwzs1"
        data-part="radio-label"
        data-scope="radio-group"
        id="radio-group:1:radio:label:1"
      >
        One
      </span>
    </label>
    <label
      class="chakra-1cqh9jq"
      data-part="radio"
      data-scope="radio-group"
      for="radio-group:1:radio:input:2"
      id="radio-group:1:radio:2"
    >
      <input
        class="chakra-radio__input chakra-0"
        data-ownedby="radio-group:1"
        data-part="radio-input"
        data-scope="radio-group"
        id="radio-group:1:radio:input:2"
        name="1"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap; word-wrap: normal;"
        type="radio"
        value="2"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control chakra-vwtg5w"
        data-part="radio-control"
        data-scope="radio-group"
        id="radio-group:1:radio:control:2"
      />
      <span
        class="chakra-radio__label chakra-4mwzs1"
        data-part="radio-label"
        data-scope="radio-group"
        id="radio-group:1:radio:label:2"
      >
        Two
      </span>
    </label>
    <label
      class="chakra-1cqh9jq"
      data-part="radio"
      data-scope="radio-group"
      for="radio-group:1:radio:input:3"
      id="radio-group:1:radio:3"
    >
      <input
        class="chakra-radio__input chakra-0"
        data-ownedby="radio-group:1"
        data-part="radio-input"
        data-scope="radio-group"
        id="radio-group:1:radio:input:3"
        name="1"
        style="border: 0px; height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap; word-wrap: normal;"
        type="radio"
        value="3"
      />
      <span
        aria-hidden="true"
        class="chakra-radio__control chakra-vwtg5w"
        data-part="radio-control"
        data-scope="radio-group"
        id="radio-group:1:radio:control:3"
      />
      <span
        class="chakra-radio__label chakra-4mwzs1"
        data-part="radio-label"
        data-scope="radio-group"
        id="radio-group:1:radio:label:3"
      >
        Three
      </span>
    </label>
  </div>
</DocumentFragment>
`;
